18/README.md

```diff
@@ -0,0 +1,119 @@
+
+We ruled this possibility out by manually monitoring the blocks written for several minutes while the test script ran. We continued to see blocks with new transactions being written well past the 10th minute of test execution, albeit only one every few seconds. The reader can repeat this test by repeatedly pinging ``$ curl localhost:46657/block | grep height`` while a test script runs and watching the block height increment.
+
+#### Intermittent Load Tests
+Next, we tested the effect of pausing the test and restarting it. In essence, rather than broadcasting at a constant rate, we set a square wave, with a constant broadcast, followed by a pause. Our assumption was that if Tendermint was experiencing a backlog somewhere, the pause should give the system a chance to clear itself out, restoring the performance once the broadcast restarted.
+
+Of greatest interest, we can see that in a series of tests where we separate broadcasts with a pause of varying duration, the transaction rate is stabilized. For example, examine lines 126 to 194 of the [data]. In this series of tests, we send 10 request batches. Each batch lasts 100 seconds, and has a request rate of 1,000 requests per second. We vary the pauses between request batch from 1 to 20 seconds in successive tests, and in all cases, our transaction rate stabilizes between 900 and 1,000 transactions per second. Compare this to sending 1,000 requests per second for 1,000 seconds straight, as we did on line 13 of the [data]. Without pauses, our average transaction rate fell to 210 transactions per second.
+
+Note that there is a sweet spot here. When we reduce our broadcast time too much, we start to lose performance, (see lines 254 - 294 in the [data]). This is due to the fact that there is a lag when we initialize the connection to Tendermint. There is a cost to breaking and re-establishing this connection that needs to be balanced against the increase in transaction rate.
+
+## Recommendations
+
+We recommend implementation of a sentinel server to field incoming requests to Tendermint. This will offer several benefits.
+
+First, we can use the sentinel to protect against DoS attacks, by limiting the number of requests from any given IP address to a reasonable number.
+
+                                            _____________________                                ____________
+    User_1, 100,000 requests/s------------>| Sentinel            |   User_1, 995 requests/s     |            |
+    User_2, 1,000 requests/s-------------->| max_req=2,000/s     |---User_2, 995 requests/s---> | Tendermint |
+    User_3, 10 requests/s----------------->| pause_interval=100s |   User_3, 10 requests/s      |____________|
+                                           |_____________________|   Break every 100s
+
+Second, we can use the sentinel to break and reinitialize connection with Tendermint as a mitigation for the perfomance degradation.
```

Comment:
> to break and reinitialize connection

You mean, to control the rate at which transactions are posted? There is no need in breaking connections.

Reply:
That was the conclusion I reached from this study. Breaking the connection to Tendermint every 100s or so mitigates the performance degradation. So if we have the sentinel, not only are we protected from DoS attacks, but we have a single point where we can manage this connection refresh.

Reply:
I mean, you do not have to break the connection to put the pressure off the system - a more efficient approach is to simply stop sending transactions.

Technically though, in order to do it, you would need to patch the `tm-bench` tool or create a custom one.

We do not have to do it now, but I would still use the term "pause the pressure" (or a similar one) instead of "break and reinitialize the connection".

Reply:
Fair point. I'll update the language to indicate that this hasn't been tested.